src/interpreter.jl

```diff
@@ -35,6 +35,14 @@ function evaluate_with_input(tab::SymbolTable, expr::Any, input::Dict)
 end
 
 
+"""
+Executes a given expression on a set of inputs and returns the respective outputs.
+"""
+function execute_program_on_examples(tab::Symboltable, expr::Any, example_inputs::Vector{Any})::Vector{Any}
```

Comment:
could we come up with shorter names? Perhaps we call the function that returns the outputs `execute` and the one that counts how many are correct `test`. It would be more convenient for use.